src/main/java/org/aksw/simba/squirrel/metadata/MetaDataHandler.java

```diff
@@ -35,18 +47,30 @@ public MetaDataHandler(String updateDatasetURI, String queryDatasetURI) {
 
     public void addMetadata(final CrawlingActivity crawlingActivity) {
         List<Triple> lstTriples = new LinkedList<>();
-        Node nodeCrawlingActivity = NodeFactory.createURI("crawlingActivity" + crawlingActivity.getId());
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.startedAtTime.asNode(), NodeFactory.createLiteral(crawlingActivity.getDateStarted())));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.endedAtTime.asNode(), NodeFactory.createLiteral(crawlingActivity.getDateEnded())));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.status.asNode(), NodeFactory.createLiteral(crawlingActivity.getState().toString())));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.wasAssociatedWith.asNode(), NodeFactory.createLiteral(String.valueOf(crawlingActivity.getWorker().getId()))));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.numberOfTriples.asNode(), NodeFactory.createLiteral(String.valueOf(crawlingActivity.getNumTriples()))));
-        Node nodeResultGraph = NodeFactory.createURI(crawlingActivity.getGraphId());
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.wasGeneratedBy.asNode(), nodeResultGraph));
-        lstTriples.add(new Triple(nodeResultGraph, MetaDataVocabulary.uriName.asNode(), NodeFactory.createURI(crawlingActivity.getCrawleableUri().getUri().toString())));
-        //TODO lstTriples.add(new Triple(nodeCrawlingActivity, NodeFactory.createURI("sq:hostedOn"), NodeFactory.createLiteral(datasetPrefix)));
-        //TODO for Meher: Merge new change from other branch manually for hadPlan
-        //lstTriples.add(new Triple(nodeCrawlingActivity,NodeFactory.createURI("prov:hadPlan"),NodeFactory.createLiteral(crawlingActivity.getHadPlan())));
+        Model model  = ModelFactory.createDefaultModel();
+        CrawleableUri uri = crawlingActivity.getCrawleableUri();
+
+        Node nodeResultGraph = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Result_"+ crawlingActivity.getGraphId(uri))));
```

Comment:
I am confused. Why are you changing the graph Id (which should be the URI of the result graph in the SPARQL store) by adding `"Result_"` and the beginning of another URI in front of it? This contradicts the idea of storing the graphId for an easy access to the graph.

As already described in my previous comments some days ago, you are not taking care of `null` values. As described above, graphId can be null.

Additionally, it is very confusing how you are using your own methods. `getUrl()` is returning a `URI` (although the name of the method says that it would be a URL). However, when you are trying to use your own method, you should see that it might be a better idea to directly create a String as result of the method, instead of investing a lot of effort into creating a `URI` object which is again transformed into a String to create a URI node.